make_prg/utils/recursive_tree_drawer.py

```diff
@@ -0,0 +1,46 @@
+from make_prg.recursion_tree import RecursiveTreeNode, SingleClusterNode
+import networkx as nx
+import matplotlib.pyplot as plt
+from pathlib import Path
+
+
+class RecursiveTreeDrawer:
+    def __init__(self, root: RecursiveTreeNode):
+        self._graph: nx.DiGraph = self._build_recursive_tree_graph(root)
+
+    @staticmethod
+    def _build_recursive_tree_graph(root: RecursiveTreeNode) -> nx.DiGraph:
+        graph = nx.DiGraph()
+        RecursiveTreeDrawer._preorder_visit(root, graph)
+        return graph
+
+    @staticmethod
+    def _preorder_visit(node: RecursiveTreeNode, graph: nx.DiGraph):
+        RecursiveTreeDrawer._visit(node, graph)
+        for child in node.children:
+            RecursiveTreeDrawer._preorder_visit(child, graph)
+
+    @staticmethod
+    def _visit(node: RecursiveTreeNode, graph: nx.DiGraph):
+        node_attributes = {}
+        node_is_SingleClusterNode = isinstance(node, SingleClusterNode)
+
+        if node.is_leaf():
+            assert node_is_SingleClusterNode, "Error, a leaf is not a SingleClusterNode"
+            prg_as_list = []
+            node._get_prg(prg_as_list)
```

Comment:
I'm guessing there is a reason that `_get_prg` fills in a list passed in as parameter rather than return a constructed list?

Reply:
(don't need to know it, just that i find the latter clearer)